vms/platformvm/blocks/builder/builder.go

```diff
@@ -194,7 +204,10 @@ func (b *builder) buildBlock() (blocks.Block, error) {
 	}
 	// [timestamp] = max(now, parentTime)
 
-	nextStakerChangeTime, err := txexecutor.GetNextStakerChangeTime(preferredState)
```

Comment:
I suggest changing this one only in tx_executor.go (and avoid passing this config value)
- jump into txexecutor.GetNextChainEventTime from txexecutor.GetNextStakerChangeTime
- add a new getter in Chain for the config value (maybe a getConfig() for later use)
- in diff, call parent state (which has access to config)